boefjes/boefjes/docker_boefjes_runner.py

```diff
@@ -0,0 +1,81 @@
+import logging
+from datetime import datetime, timezone
+
+import docker
+from docker.errors import APIError, ContainerError, ImageNotFound
+from requests import HTTPError
+
+from boefjes.clients.bytes_client import BytesAPIClient
+from boefjes.clients.scheduler_client import SchedulerAPIClient, TaskStatus
+from boefjes.config import settings
+from boefjes.job_models import BoefjeMeta
+from boefjes.katalogus.models import Boefje
+
+logger = logging.getLogger(__name__)
+
+
+class DockerBoefjesRunner:
+    def __init__(self, boefje_resource: Boefje, boefje_meta: BoefjeMeta):
+        self.boefje_resource = boefje_resource
+        self.boefje_meta = boefje_meta
+        self.docker_client = docker.from_env()
+        self.scheduler_client = SchedulerAPIClient(settings.scheduler_api)
+        self.bytes_api_client = BytesAPIClient(
+            settings.bytes_api,
+            username=settings.bytes_username,
+            password=settings.bytes_password,
+        )
+
+    def run(self) -> None:
```

Comment:
If you implement the same return types of the `BoefjeJobRunner` class here, you can:
- Only return the mime_types and the raw logs (Looks like the return type of `BoefjeJobRunner.run()` needs to be updated to the one from `LocalBoefjeJobRunner.run()` also)
- Let the JobHandler save the logs to bytes
- Let the `SchedulerWorkerManager` take care of updating the scheduler task status. You can simply raise the right errors here and it should be taken care of there.

Reply:
I'll check if I can make that work. The `JobHandler` doesn't work that great with the asynchronous design, because it overwrites the status that the output API endpoint already saved to Bytes. We could temporarily save the data from the output API endpoint somewhere else, or read it from Bytes to then return it to the `JobHandler`, but that feels like we're working against the abstractions we created ourselves.

Reply:
If it is a lot of work I don't think it is worth the effort. I don't think we want to keep the current boefje runner long term, because even when we want to continue execute multiple boefjes in the same proces that should run in its own container and not in the runner container. Then the new way of doing this would be the only way. It wouldn't make much sense to spend a lot of time trying to fit the new code into the old abstractions if we eventually will get rid of the old code.